File: front_end/src/service/DanhMucService.js
import axios from 'axios';

const REST_API_BASE_URL = 'http://localhost:8080/danh-muc'

export const listDanhMuc = (pageable) => {
    return axios.get(REST_API_BASE_URL, {
        params: { page: pageable.page, size: pageable.size }
    });
};

export const createDanhMuc = (danhMuc) => axios.post(REST_API_BASE_URL, danhMuc);
export const getDanhMuc = (danhMucId) => axios.get(REST_API_BASE_URL + '/' + danhMucId);
export const updateDanhMuc = (danhMucId, danhMuc) => axios.put(REST_API_BASE_URL + '/' + danhMucId, danhMuc);
export const deleteDanhMuc = (danhMucId) => axios.delete(REST_API_BASE_URL + '/' + danhMucId);

File: src/main/java/com/example/shoe/service/DanhMucService.java
package com.example.shoe.service;

import com.example.shoe.dto.request.DanhMucRequest;
import com.example.shoe.dto.response.DanhMucResponse;
import com.example.shoe.entity.DanhMuc;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface DanhMucService {
    public Page<DanhMuc> getDanhMuc(Pageable pageable);
    public DanhMuc createDanhMuc(DanhMucRequest request);
    public DanhMucResponse getDanhMucId(Integer id);
    public DanhMucResponse updateDanhMuc(Integer id, DanhMucRequest request);
    public void deleteDanhMuc(Integer id);
    public Page<DanhMuc> searchDanhMuc(String keyword, Pageable pageable);
    
}
